### PR TITLE
Add Postgres DB Schema on service startup.

### PR DESCRIPTION
**Pull request by mike-kelly-methods**
### What

Adds the required Postgres Schema on service startup.

### How to review

Code looks sensible, build and tests pass.

### Who can review

!me

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -28,6 +29,13 @@ type Config struct {
 	RDSDBInstance1             string `envconfig:"RDSINSTANCE1"`
 	RDSDBInstance2             string `envconfig:"RDSINSTANCE2"`
 	RDSDBInstance3             string `envconfig:"RDSINSTANCE3"`
+	// flag to use local postres instace provided by dp-compose
+	DPPostgresLocal            bool   `envconfig:"USEPOSTGRESLOCAL"`
```
Comment (Marianasilvago) [inline config/config.go:33]: looks like we need these to for dev testing, can we see if we can existing environment variable with defaults configured here: https://github.com/ONSdigital/dp-areas-api/blob/develop/config/config.go#L55

Comment (mike-kelly-methods) [reply config/config.go:33]: 👍

File: models/database_schema_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		})
+
+		Convey("Ensure execution list built successfully", func() {
+			databaseSchema := models.DatabaseSchema{
+				DBName: "dp-areas-api",
+				SchemaString: DBRelationalSchema.DBSchema,
+			}
+			_ = databaseSchema.BuildDatabaseSchemaModel()
+			databaseSchema.TableSchemaBuilder()
+			So(len(databaseSchema.ExecutionList), ShouldBeGreaterThanOrEqualTo, 5)
```
Comment (Marianasilvago) [inline models/database_schema_test.go:63]: Can we just also assert on the actual query if possible.

File: config/config.go [hunk 3]
```diff
@@ -70,3 +78,18 @@ func Get() (*Config, error) {
...
+}
+
+// GetLocalDBConnectionString returns local connection string
+func (c Config) GetLocalDBConnectionString () string {
+	return fmt.Sprintf("postgres://%s:%s@localhost:%s/%s", c.DPPostgresUserName, c.DPPostgresUserPassword, c.DPPostgresLocalPort, c.DPPostgresLocalDB)
+}
+
+// GetRemoteDBConnectionString returns remote connection string
+func (c Config) GetRemoteDBConnectionString (authToken string) string {
+	return fmt.Sprintf("host=%s port=%s user=%s password=%s dbname=%s", c.RDSDBHost, c.RDSDBPort, c.RDSDBUser, authToken, c.RDSDBName)
```
Comment (Marianasilvago) [inline config/config.go:94]: is it possible to use similar connection string for both local and remote DB connection?

Comment (mike-kelly-methods) [reply config/config.go:100]: Unfortunately not - the PGX Pool connections to RDS are using IAM authentication, using an auth token generated by `github.com/aws/aws-sdk-go/service/rds/rdsutils`. The local connection is using a plain old connection string with the db name and password.

**Review by Marianasilvago — APPROVED** [review]

**Pull request merged**